leakcanary-deobfuscation-gradle-plugin/src/main/java/com/squareup/leakcanary/deobfuscation/CopyObfuscationMappingFileTask.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.leakcanary.deobfuscation
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+@CacheableTask
+open class CopyObfuscationMappingFileTask : DefaultTask() {
+
+  @Input
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mappingFile: File? = null
```

Comment:
Is this configurable? How does one go abou configuring it?

Reply:
Not sure what do you mean but I'll try to reply. This file is passed to the task by our plugin. It shouldn't be changed by the user. I don't know if it's possible but I think if someone is able to get a reference to our task from outside of our plugin then it's possible to pass any file you want here. But then again - our plugin should override it when it's being executed. @stephanenicolas do you have something to add here?

Reply:
Makes sense! Sorry I'm not super up to date on gradle